src/Components/Home.js

```diff
@@ -0,0 +1,68 @@
+import { useEffect } from 'react';
+import { Button, Card, Container } from 'react-bootstrap';
+import { useDispatch, useSelector } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { NavLink } from 'react-router-dom';
+import { fetchCountries, filteredCountries } from '../Redux/reducers/countries';
+
+const Home = () => {
+  let countryName;
+
+  const dispatch = useDispatch();
+  const countries = useSelector((state) => state.countryReducer);
+
+  useEffect(() => {
+    dispatch(fetchCountries());
+  }, [dispatch]);
+
+  return (
+    <Container>
+      <div className="countrySearchBox">
+        <input className="countrySearch" type="text" placeholder="Country Name" value={countryName} onChange={(e) => { countryName = (e.target.value); }} />
+        <Button
+          variant="secondary"
+          type="button"
+          onClick={async () => {
+            await dispatch(fetchCountries());
+            dispatch(filteredCountries(countryName));
+          }}
+        >
+          Search
+        </Button>
+      </div>
```

Comment:
- [ ] Your filter work perfectly :+1: however, we are asked to use the **Filter** stateless component to filter data retrieved from the API. Let's kindly have a Filter stateless component on its own to carry out the filtering.